core/templates/components/forms/schema-based-editors/schema-based-custom-editor.component.spec.ts

```diff
@@ -16,23 +16,51 @@
  * @fileoverview Unit tests for schema-based editor component for custom values
  */
 
-import { NO_ERRORS_SCHEMA } from '@angular/core';
-import { FormControl, FormsModule } from '@angular/forms';
-import { ComponentFixture, fakeAsync, TestBed, waitForAsync } from '@angular/core/testing';
+import { Component, EventEmitter, NO_ERRORS_SCHEMA, forwardRef } from '@angular/core';
+import { ControlValueAccessor, FormControl, FormsModule, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { ComponentFixture, fakeAsync, flush, TestBed, tick, waitForAsync } from '@angular/core/testing';
 import { SchemaBasedCustomEditorComponent } from './schema-based-custom-editor.component';
 import { SchemaDefaultValue } from 'services/schema-default-value.service';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { TranslateFakeLoader, TranslateLoader, TranslateModule, TranslateService } from '@ngx-translate/core';
+
+@Component({
+  selector: 'object-editor',
+  template: '',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => MockObjectEditorComponent),
+      multi: true
+    }
+  ]
+})
+export class MockObjectEditorComponent implements ControlValueAccessor {
+  writeValue(value: string | number | null): void {}
+  registerOnChange(fn: () => void): void {}
+  registerOnTouched(fn: (() => void) | undefined): void {}
+  setDisabledState?(isDisabled: boolean): void {}
+}
 
 describe('Schema Based Custom Editor Component', () => {
   let component: SchemaBasedCustomEditorComponent;
   let fixture: ComponentFixture<SchemaBasedCustomEditorComponent>;
 
   beforeEach(waitForAsync(() => {
     TestBed.configureTestingModule({
-      imports: [FormsModule],
+      imports: [FormsModule, HttpClientTestingModule,
+        TranslateModule.forRoot({
+          loader: {
+            provide: TranslateLoader,
+            useClass: TranslateFakeLoader
+          }
+        })],
       declarations: [
-        SchemaBasedCustomEditorComponent
+        SchemaBasedCustomEditorComponent,
+        MockObjectEditorComponent
```

Comment:
Add comment explaining why this is needed

Reply:
Added it to the place where the mock is defined.